3party/opening_hours/opening_hours.cpp

```diff
@@ -103,6 +111,12 @@ void PrintPaddedNumber(std::ostream & ost, TNumber const number, uint32_t const
   ost << std::setw(padding) << std::setfill('0') << number;
 }
 
+template <typename TNumber, typename std::enable_if<IsChar(TNumber{}), void*>::type = nullptr>
+void PrintPaddedNumber(std::ostream & ost, TNumber const number, uint32_t const padding = 1)
+{
+  PrintPaddedNumber(ost, static_cast<uint32_t>(number), padding);
```

Comment:
Зачем вообще так сложно? Юра, почему ты против простого решения без потоков?

> On Mar 23, 2016, at 00:14, ygorshenin <notifications@github.com> wrote:
> 
> In 3party/opening_hours/opening_hours.cpp <https://github.com/mapsme/omim/pull/2422#discussion_r57070270>:
> 
> > @@ -103,6 +111,12 @@ void PrintPaddedNumber(std::ostream & ost, TNumber const number, uint32_t const
> >    ost << std::setw(padding) << std::setfill('0') << number;
> >  }
> >  
> > +template <typename TNumber, typename std::enable_if<IsChar(TNumber{}), void*>::type = nullptr>
> > +void PrintPaddedNumber(std::ostream & ost, TNumber const number, uint32_t const padding = 1)
> > +{
> > +  PrintPaddedNumber(ost, static_cast<uint32_t>(number), padding);
> Это неправильно знаковый char конвертировать в uint. Используй просто int.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub <https://github.com/mapsme/omim/pull/2422/files/aa6b2c9bbf18c470b7defcdda2ff6163fc138eda#r57070270>